backend/src/apiserver/server/job_server.go

```diff
@@ -403,33 +399,38 @@ func (s *JobServer) canAccessJob(ctx context.Context, jobID string, resourceAttr
 		// Skip authorization if not multi-user mode.
 		return nil
 	}
-
+	namespace := ""
 	if len(jobID) > 0 {
 		job, err := s.resourceManager.GetJob(jobID)
 		if err != nil {
-			return util.Wrap(err, "Failed to authorize with the job ID")
+			return util.Wrap(err, "failed to authorize with the recurring run ID")
 		}
-		if len(resourceAttributes.Namespace) == 0 {
-			if len(job.Namespace) == 0 {
-				return util.NewInternalServerError(
-					errors.New("Empty namespace"),
-					"The job doesn't have a valid namespace",
-				)
+		if job.Namespace == "" {
+			experiment, err := s.resourceManager.GetExperiment(job.ExperimentId)
+			if err != nil {
+				return util.NewInvalidInputError("recurring run %v has an empty namespace and the parent experiment %v could not be fetched: %s", jobID, job.ExperimentId, err.Error())
```

Comment:
Why would a recurring run has no parent experiment? I think that would be a case of internal server error.

Reply:
Both `namespace` and `experimentId` are coming from the user, so `NewInvalidInputError` makes sense when someone specifies a non-existing id, for example. But, that shouldn't happen. Let me change it to the server error.